release-notes.md

```diff
@@ -203,6 +212,79 @@
     - [Unit testing](#unit-testing)
     - [Integration tests parallelization](#integration-tests-parallelization)
 
+# v24.0
+
+This release has been tracked through the [v24.0
+project](https://github.com/cloud-hypervisor/cloud-hypervisor/projects/28).
+
+### Bypass Mode for `virtio-iommu`
+
+`virtio-iommu` specification describes how a device can be attached by default
+to a bypass domain. This feature is particularly helpful for booting a VM with
+guest software which doesn't support `virtio-iommu` but still need to access
+the device. Now that Cloud Hypervisor supports this feature, it can boot a VM
+with Rust Hypervisor Firmware or OVMF even if the `virtio-block` device exposing
+the disk image is placed behind a virtual IOMMU.
+
+### Ensure Identifiers Uniqueness
+
+Multiple checks have been added to the code to prevent devices with identical
+identifiers to be created, and therefore avoid unexpected behaviors at boot or
+whenever a device was hot plugged into the VM.
+
+### Sparse Mmap support
+
+Adding sparse mmap support to both VFIO and vfio-user devices. This allows the
+device regions that are not fully mappable to be partially mapped. And the more
+a device region can be mapped into the guest address space, the less VM exits
+will be generated when this device is accessed. This directly impacts the
+performance related to this device.
+
+### Expose Platform Serial Number
+
+Adding a new `serial_number` option to `--platform`, allowing a user to set a
+specific serial number for the platform. This number is exposed to the guest
+through the SMBIOS.
+
+### Notable Bug Fixes
+
+* Fix loading RAW firmware (#4072)
+* Reject compressed QCOW images (#4055)
+* Reject virtio-mem resize if device is not activated (#4003)
+* Fix potential mmap leaks from VFIO/vfio-user MMIO regions (#4069)
+* Fix algorithm finding HOB memory resources (#3983)
+
+### Notable Improvements
+
+* Refactor interrupt handling (#4083)
+* Load kernel asynchronously (#4022)
+* Only create ACPI memory manager DSDT when resizable (#4013)
+
```

Comment:
Include a section for the deprecations as they will be removed in the next release cycle.